src/_check.sh

```diff
@@ -18,10 +18,11 @@ function mod_check() {
         error "Pending Changes:"
         msg "    %s\n" "$pending"
     fi
-    VALIDATION_ONLY=1 configure_couchbase_users
-    postgres_metrics_validation
-    check_for_maintenance_mode
-
+    if [ ${INSTALL_ONLY:-0} -eq 0 ]; then
+      VALIDATION_ONLY=1 configure_couchbase_users
```

Comment:
What about moving this logic up a layer? eg, just put the user creation stuff into the install function. Then drop the `mod_check` call from `mod_install`

Reply:
edit: I got confused, I mean _don't_ wrap this stuff in logic to determine if an install is running or not, just have `mod_install` in the _pre-install_ step do docker compose validation instead of calling the `mod_check` function. That way it is doing less work and we don't have branching logic.

Reply:
The preinstall flag is never used IIRC. When the instance is stood up, `plextrac --preinstall` is never called and in the normal workflow it isn't called either. In fact its not used outside of this context:
```bash
# only used for mod_check
      "--pre" | "--preinstall")
        DO_PREINSTALL_CHECKS=1
        shift
        ;;
```

This also means that these two functions don't get run:

```bash
if [ ${DO_PREINSTALL_CHECKS:-0} -eq 1 ]; then
    title "Running pre-installation checks"
    _check_system_meets_minimum_requirements
    _check_no_existing_installation
  else
    title "Running checks on installation at '${PLEXTRAC_HOME}'"
```

Regardless of that, I've copied the compose file validation to the `mod_install` function so it will validate that before starting the DBs

Reply:
Yep, the `preinstall` checks are for "readiness" of a host system and would be manually run by the end user fysa